src/Features/Core/Portable/CodeRefactoringHelpers.cs

```diff
@@ -3,12 +3,45 @@
 using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis
 {
     internal static class CodeRefactoringHelpers
     {
+        /// <summary>
+        /// <para>
+        /// Returns an instance of <typeparamref name="TSyntaxNode"/> for refactoring given specified selection in document or null
+        /// if no such instance exists.
+        /// </para>
+        /// <para>
+        /// A <typeparamref name="TSyntaxNode"/> instance is returned if selection is zero-width and inside/touching a Token whose
+        /// direct parent is of type <typeparamref name="TSyntaxNode"/> or if a Node of said type is the smallest Node containing
+        /// the whole <paramref name="selection"/>. Otherwise returns <code>null</code>.
+        /// </para>
+        /// <para>
+        /// Note: this function strips all whitespace from both the beginning and the end of given <paramref name="selection"/>.
+        /// The stripped version is then used to determine relevant <see cref="SyntaxNode"/>.
+        /// </para>
+        /// </summary>
+        public static async Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(
+            Document document, TextSpan selection, CancellationToken cancellationToken, bool traverseUpInSyntaxTreeIfOnEdgeOfSpan = false) where TSyntaxNode : SyntaxNode
```

Comment:
For traverseUpInSyntaxTreeIfOnEdgeOfSpan , i personally think convert-local-function refactoring should pass true for this